plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -82,6 +90,8 @@
   @VisibleForTesting int adminPort = -1;
 
   private DevAppServerOutputListener serverOutputListener;
+  
+  private Map<String, String> moduleToUrlMap = Maps.newHashMap();
```

Comment:
One thing I noticed is that this map is created once when a server is first created. That is, restarting a server will carry the values extracted from the previous run. Most of the time this will be okay because the values will be overwritten, but I see a slight chance of this going out of sync when a user modifies `appengine-web.xml`. So maybe it'd better to reset this map at the start of a server.